### ISSUE TITLE
compute_km_weights args not all documented

### ISSUE THREAD

**Issue by JayAchar** [nyilin/SamplingDesignTools#1]
I've just been trying to use this function to generate some KM-type weights :)

I came across a couple of challenges I thought it would be worth documenting. 

My use-case is to try and use a data frame of already sampled NCC data. I have the numbers of subjects at risk in each riskset from the original cohort. 

# Reprex
 
```
# Dummy data
ncc <- data.frame(id = 1:6,
           Set = rep(c(1:3), each = 2),
           Time = rep(c(1:3), each = 2),
           Fail = rep(c(0, 1), 3))

risk_table <- data.frame(Time = 1:3,
           n.risk = c(6, 4, 2))

# Call function
SamplingDesignTools::compute_km_weights(
  ncc = ncc,
  n_at_risk = risk_table,
  t_match_name = "Time",
  n_per_case = 1
)
```

# Problem 

First, I seem to get errors for parameters that I don't think are required as per the help file. Then, when I enter them, the function errors because I have entered them. 

When I use `debug()` to work through the function, I see that the args - `id_name` and `set_id_name` are not mentioned in the documentation for the function. 

When I add these values, I think I have the function working, but I'm a little unsure of the output. The KM weight for the subject with id == 1 is Inf. Since there are n.risk=6 at this time, I would have expected a smaller value. 

If I have time over the weekend, I'll try and take a closer look at the code and submit a pull request :)

**Comment by nyilin** [comment]
Thank you for your comment. I have modified the documentation for `compute_km_weights` to be more specific about the input parameters in different use cases, and have added error messages to help users find out if they are missing any important input. 

I would also like to highlight a message that was not clear from the previous version of documentation for the benefit of new users who happens to see this post later. The dummy data you have generated is sufficient for debugging, but in real data applications when only the NCC sample is available, it is in fact insufficient to only have the variable `Time`, which is the time of the event in each matched set. There needs to be another column that stores the actual time of event or censoring for each subject, so that the correct weights can be computed for the sampled controls, and subsequent weighted analysis can be performed appropriately by analysing this actual time of event/censoring. 

I have corrected the definition of a risk set when controls are censored at exactly the same time of an event, so that the weight is now correct for subject 1 in your example.
